src/com/photoncat/aiproj2/heuristic/Heuristic2.java

```diff
@@ -0,0 +1,902 @@
+                    score += 25000;
+                    if (player==PieceType.CIRCLE){
+                        score+=50000;
+                    }
+                    continue;
+                }
+                if ((getByDirLen(board, row, col, i, -2) == opp ||
+                        getByDirLen(board, row, col, i, -2) == -1) && (
+                        getByDirLen(board, row, col, i, 2) == opp ||
+                                getByDirLen(board, row, col, i, 2) == -1)) {
+                    continue;
+                } else {
+                    score += 700;
+                    continue;
+                }
+            }
+            if (getByDirLen(board, row, col, i, -1) == cur &&
+                    getByDirLen(board, row, col, i, -2) == 0
+                    && getByDirLen(board, row, col, i, 1) == 0
+                    ) {
+                one++;
+            }
+            int lonewolf=0;
+            for(int k=-4;k<=0;k++){
+                int temp=0;
+                for (int l = 0; l <= 4; l++) {
+                    if (getByDirLen(board,row,col,i,k+l)==cur){
+                        temp++;
+                    }
+                    else {
+                        if (    getByDirLen(board,row,col,i,k+l)==opp||
+                                getByDirLen(board,row,col,i,k+l)==-1)
+                            temp=0;
+                        break;
+                    }
+                }
+                lonewolf+=temp;
+            }
+            score+=lonewolf*15;
+
+
+        }
+        if (one>=2){
+            score+=20000;
+            if (player==PieceType.CIRCLE){
+                score+=500;
+            }
+        }
+        return score;
+    }
+
+    private int getByDirLen(Board board, int row, int col, int dir, int len) {
+        //len typically is negative.
+        //up
+        if (dir==1){
+            row+=len;
+        }
+        //Northwest
+        else if (dir==2){
+            row+=len;
+            col+=len;
+        }
+        //left
+        else if (dir==3){
+            col+=len;
+        }
+        //Southwest
+        else if (dir==4){
+            row-=len;
+            col+=len;
+        }
+        //down
+        else if (dir==5){
+            row-=len;
+        }
+        //Southeast
+        else if (dir==6){
+            row-=len;
+            col-=len;
+        }
+        //right
+        else if (dir==7){
+            col-=len;
+        }
+        //Northeast
+        else if (dir==8){
+            row+=len;
+            col-=len;
+        }
+        int N=board.getSize();
+        if (row<0||col<0||row>=N||col>=N){
+            return -1;
+        }
+        if (board.getPiece(row,col)==PieceType.CROSS){
+            return 1;
+        }
+        else if (board.getPiece(row,col)==PieceType.CIRCLE){
+            return 2;
+        }
+        else {
+            return 0;
+        }
+    }
+    private boolean checkWin(Board board,PieceType player) {
```

Comment:
Again you don't need this function. Board provides a wins() function and it actually caches that result so you don't have to count pieces yourself.